File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        // int[] answer = {};
        List<Integer> answer = new ArrayList<>();
        int n = progresses.length;
        int[] days = new int[n];
        for(int i = 0 ; i < n ; i++){
            days[i] = (int)Math.ceil((double)(100 - progresses[i])/speeds[i]);
        }
        int c = 1, prev = days[0];
        for(int i = 1 ; i < n ; i++){
            if(prev >= days[i]){
                c += 1;
            } else {
                answer.add(c);
                c = 1;
                prev = days[i];
            }
        }
        answer.add(c);
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    # 남은 일 수를 우선 계산
    # [7,3,9]
    days = [0] * len(speeds)
    for i in range(len(speeds)):
        days[i] = math.ceil((100 -progresses[i]) / speeds[i])
        
    m = days[0]
    c = 1
    # if len(days) == 1:
    #     return [1]
    for i in range(1, len(days)):
        if m >= days[i]:
            c += 1
        else:
            answer.append(c)
            m = days[i]
            c = 1
        if i == len(days) -1 :
            answer.append(c)
    return answer